torchmetrics/utilities/data.py

```diff
@@ -16,6 +16,17 @@
 import torch
 from torch import Tensor, tensor
 
+from torchmetrics.utilities.imports import _TORCH_GREATER_EQUAL_1_6, _TORCH_GREATER_EQUAL_1_7, _TORCH_GREATER_EQUAL_1_8
+
+if _TORCH_GREATER_EQUAL_1_8:
+    deterministic = torch.are_deterministic_algorithms_enabled
+elif _TORCH_GREATER_EQUAL_1_7:
+    deterministic = torch.is_deterministic
+elif _TORCH_GREATER_EQUAL_1_6:
+    deterministic = torch._is_deterministic
+else:
+    deterministic = lambda: False
```

Comment:
```suggestion
    def deterministic() -> bool:
        return True
```

from [pep8](https://peps.python.org/pep-0008/#programming-recommendations): 
> Always use a def statement instead of an assignment statement that binds a lambda expression directly to an identifier